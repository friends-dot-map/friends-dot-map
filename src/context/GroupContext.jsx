import { createContext, useContext, useState, useEffect } from 'react';
import { getAllProfiles } from '../services/profiles';

export const GroupContext = createContext();

const GroupProvider = ({ children }) => {
  const [group, setGroup] = useState([]);

  useEffect(() => {
    const fetchGroup = async () => {
      try {
        const data = await getAllProfiles();
        setGroup(data);
      } catch (error) {
        throw error('something went horribly wrong');
      }
    };
    fetchGroup();
  }, []);
  const groupValue = { group, setGroup };
  return (
    <GroupContext.Provider value={groupValue}>{children}</GroupContext.Provider>
  );
};

const useGroup = () => {
  const context = useContext(GroupContext);

  if (context === undefined) {
    throw new Error('useGroup must be used within a GroupProvider');
  }

  return context;
};

export { GroupProvider, useGroup };
