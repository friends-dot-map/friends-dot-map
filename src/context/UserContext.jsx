import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
