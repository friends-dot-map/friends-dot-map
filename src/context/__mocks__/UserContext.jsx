import { createContext, useContext, useState } from 'react';

export const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const userValues = { user, setUser };

  return (
    <UserContext.Provider value={userValues}>{children}</UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
