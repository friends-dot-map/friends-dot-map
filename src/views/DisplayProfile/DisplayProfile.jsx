
export default function DisplayProfile() {
  return (
    <div>DisplayProfile</div>
  )
}
