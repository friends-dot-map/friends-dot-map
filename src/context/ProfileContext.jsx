import { createContext, useContext, useState, useEffect } from 'react';
import { getProfile } from '../services/profiles';
import { useUser } from './UserContext';
import { formatDate } from '../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [loading, setLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = await getProfile(user.email);
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    loading,
    setLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
