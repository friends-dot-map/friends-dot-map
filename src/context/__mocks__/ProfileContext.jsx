import { createContext, useContext, useState, useEffect } from 'react';
import { useUser } from '../UserContext';
import { formatDate } from '../../utils/utils';

export const ProfileContext = createContext();

const ProfileProvider = ({ mockProfile, children }) => {
  const [userCoords, setUserCoords] = useState({ latitude: '', longitude: '' });
  const [profile, setProfile] = useState({
    username: '',
    first_name: '',
    likes: '',
    status: '',
    avatar: '',
    coords: userCoords,
    updated_at: formatDate(),
  });
  const [profileLoading, setProfileLoading] = useState(true);
  const { user } = useUser();

  useEffect(() => {
    const fetchProfile = async () => {
      try {
        const profileData = mockProfile || {};
        setProfile(profileData);
      } catch (error) {
        setProfile({});
      }
      setProfileLoading(false);
    };
    fetchProfile();
  }, [user]);

  const profileValues = {
    profile,
    profileLoading,
    setProfile,
    userCoords,
    setUserCoords,
  };

  return (
    <ProfileContext.Provider value={profileValues}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(ProfileContext);

  if (context === undefined) {
    throw new Error('useProfile must be used within a ProfileProvider');
  }

  return context;
};

export { ProfileProvider, useProfile };
