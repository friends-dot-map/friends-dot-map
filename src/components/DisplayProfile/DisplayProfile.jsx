import { useProfile } from '../../context/ProfileContext';
import { useState } from 'react';
import { updateStatus } from '../../services/profiles';
import { Link, useHistory, useParams } from 'react-router-dom';
import { useGroup } from '../../context/GroupContext';

export default function DisplayProfile() {
  const [statusEdit, setStatusEdit] = useState(false);
  const [newStatus, setNewStatus] = useState('');
  const {
    loading,
    userCoords,
    profile: { user_id },
    setProfile,
  } = useProfile();
  const params = useParams();
  const history = useHistory();
  const { group } = useGroup();


  const [currentProfile] = group.filter(
    (user) => user.username === params.username
  );

  const handleStatus = async () => {
    try {
      const [data] = await updateStatus(newStatus, userCoords, user_id);
      console.log(data);
      setProfile((prevState) => ({
        ...prevState, status: data.status, coords: data.coords
      }))
      console.log(profile);
      setStatusEdit(false);
    } catch (error) {
      throw new Error('Was not able to update status');
    } 
      history.push('/');
    
    
  };

  if (loading) return <h1>Loading....</h1>;

  return (
    <div>
      <p>{currentProfile.avatar}</p>
      <p>{currentProfile.username}</p>
      <p>{currentProfile.first_name}</p>
      <p>{currentProfile.likes}</p>
      {statusEdit ? (
        <>
          <input
            value={newStatus}
            onChange={(e) => setNewStatus(e.target.value)}
            type="text"
          />
          <button onClick={handleStatus}>Post Status</button>
        </>
      ) : (
        <>
          <p>{currentProfile.status}</p>
          <button
            onClick={() => {
              setStatusEdit(true);
            }}
          >
            Update Status
          </button>
        </>
      )}
      <button>
        <Link to="/edit">Edit Profile</Link>
      </button>
    </div>
  );
}
