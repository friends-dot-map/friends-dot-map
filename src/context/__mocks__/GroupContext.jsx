import { createContext, useContext, useState, useEffect } from 'react';
import { useProfile } from '../ProfileContext'


export const GroupContext = createContext();

const GroupProvider = ({ mockGroup, children }) => {
  const [group, setGroup] = useState([]);
  const { profile } = useProfile();

  useEffect(() => {
    const fetchGroup = async () => {
      try {
        const data = mockGroup || [];
        setGroup(data);
      } catch (error) {
        throw error('something went horribly wrong');
      }
    };
    fetchGroup();
  }, [profile]);

  const groupValue = { group };

  return (
    <GroupContext.Provider value={groupValue}>{children}</GroupContext.Provider>
  );
};

const useGroup = () => {
  const context = useContext(GroupContext);

  if (context === undefined) {
    throw new Error('useGroup must be used within a GroupProvider');
  }

  return context;
};

export { GroupProvider, useGroup };
