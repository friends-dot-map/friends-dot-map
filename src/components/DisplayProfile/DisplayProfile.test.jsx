import DisplayProfile from "./DisplayProfile";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom/cjs/react-router-dom.min";
import { ProfileProvider } from "../../context/ProfileContext";
import { GroupProvider } from "../../context/GroupContext";
import { UserProvider } from "../../context/UserContext";

test('auth form renders login page', () => {
    const { container } = render (
        <UserProvider>
            <ProfileProvider>
                <GroupProvider>
                    <MemoryRouter>
                        <DisplayProfile />
                    </MemoryRouter>
                </GroupProvider>
            </ProfileProvider>
        </UserProvider>
    
    );
    

    expect(container).toMatchSnapshot();
});