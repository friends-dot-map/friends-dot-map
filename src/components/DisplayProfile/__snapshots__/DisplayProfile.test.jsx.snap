// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display profile renders 1`] = `
<div>
  <div
    class="flex flex-col text-center items-center justify-evenly p-2 h-3/4"
  >
    <p
      class="text-8xl md:text-9xl"
    >
      üê∂
    </p>
    <div
      class="space-y-4"
    >
      <h2
        class="text-5xl md:text-7xl font-cursive tracking-wider"
      >
        spongebob
      </h2>
      <p
        class="text-lg md:text-2xl"
      >
        (
        charles
        )
      </p>
    </div>
    <label
      class="text-2xl md:text-4xl font-cursive tracking-wider"
      for="status"
    >
      Current Status
    </label>
    <p
      class="text-lg md:text-2xl italic"
      id="status"
    >
      rockin in the free world
    </p>
    <p
      class="text-sm md:text-lg"
      id="updated"
    >
      posted at 
    </p>
    <button
      class="bg-white/0 text-teal w-52 text-sm md:text-base p-2 rounded-md"
    >
      Clear status and location
    </button>
    <button
      class="bg-teal text-white w-52 md:text-2xl p-2 rounded-md"
    >
      Update Status
    </button>
    <label
      class="text-2xl md:text-4xl font-cursive tracking-wider"
      for="likes"
    >
      Likes
    </label>
    <p
      class="text-lg md:text-2xl"
      id="likes"
    >
      snails
    </p>
    <a
      href="/edit"
    >
      <button
        class="bg-white text-teal  ring-tint p-2 rounded-md"
      >
        Edit Profile
      </button>
    </a>
    <a
      href="/"
    >
      <button
        class=" bg-teal text-white p-2 rounded-md"
      >
        Back to Map
      </button>
    </a>
    <button
      class="bg-orange text-white  ring-tint p-2 rounded-md"
    >
      Delete Profile
    </button>
  </div>
</div>
`;
